src/components/habit-tracker/HabitList.tsx

```diff
@@ -131,6 +130,7 @@ export function HabitList({
       handleLog={handleLog}
       handleRemoveLog={handleRemoveLog}
       habitsLog={habitsLog}
+      isFutureDate={isFutureDate}
```

Comment:
what I meant by disabling for future date was, to not allow the user to even select the date in the future. Right now, I am able to select a date in the future but the logging habit button is disabled. It would confuse the user that why am I not able to log these habits.